### DISCUSSION TITLE
Layer color override

### DISCUSSION THREAD

**Discussion by wandrinsheep** [keiwando/toonsquid#427] [Ideas]
Hi there, I ran a search for this first before making this thread, hopefully I’ve not duplicated. What I’d like to ask for is a layer color override feature. I work quick and sloppy in the animatic/layout stages and I’d like to be able to just quickly  override colors on any layer for readability and separation  of features. Preferably this should available on any layer and maybe layer groups. 

I appreciate your app greatly. And how much work you’ve put into this. Thanks for hearing my thoughts.

**Comment by Cat-Butler** [comment]
Hi @wandrinsheep - just for my own understanding: 

Do you want to override a single colour layer e.g. change the colour of a monochrome line art layer which, if I'm thinking correctly,  is doable now with some layer masking (in the timeline, single layers or groups), or override multiple colours (e.g. override the yellow fill areas with green in all drawings on this layer, and the red-filled areas with blue) which will probably require some serious work by the developer? 

If the latter, I'll definitely add my vote (great idea), although I'm not really expecting it particularly soon.

**Reply by wandrinsheep** [reply]
Actually it was the former, didn’t even realise the mask option was there on the timeline, thank you. But also now that you’ve mentioned the latter that would indeed be fantastic but it would be a problematic undertaking with lots of edge cases(like anti aliasing, color thresholds etc). I’d be more interested for the short term in multiple layer fill, which, if you’ve ever used toonboom comes quite in handy for helping to save time on the coloring process. It essentially fills all the layers(or a specified range) with a color you’ve chosen on all the layers within the scope that are capable of being filled(enclosed) at that  selected area.

**Reply by Cat-Butler** [reply]
Possibly similar requests if you want to upvote:
https://github.com/keiwando/toonsquid/discussions/151
https://github.com/keiwando/toonsquid/discussions/404

I'm more Moho-influenced myself, so personally I'd rather see improved rigging features (fill it once then manipulate the filled shape/area), but I have no objections to feature improvements for the frame-by-frame animators.
